### ISSUE TITLE
Buttons that have visible text should not use tooltips

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#42676]
### Description

In the Site Editor, when the preference 'Show button text labels' is enabled, some buttons show text instead of icons. So far so good. However, these buttons still use a tooltip that appears on hover and focus. The tooltip just repeats the visible button text and:
- it clutters the UI
- it's redundant
- it's inconsistent with the Post Editor, where the tooltips appear only when buttons show icons (with some exceptions, see below)

Screenshots from the Site Editor, while focusing those buttons:

<img width="410" alt="Screenshot 2022-07-25 at 15 34 54" src="https://user-images.githubusercontent.com/1682452/180796879-07778352-d4d0-4c68-a35d-f7e01a9eb955.png">

### Step-by-step reproduction instructions

- Go to the Side Editor
- Enable 'Show button text labels' under Options > Preferences.
- Hover or focus the buttons in the right part of the top bar.
- Observe the buttons do have tooltips that repeat the visible text.
- Repeat the steps above in the Post Editor.
- Observe the buttons do not have tooltips.

**Exceptions and inconsistencies:**

Only buttons that are associated with a keyboard shortcut always show a tooltip. For example: Undo, Redo, and List View. I guess this is meant to actually display the keys of the shortcut but I'm not sure it's a great pattern because these buttons behavior differs from the one of other buttons and it may be unexpected or confusing for users. Screenshot:

<img width="413" alt="Screenshot 2022-07-25 at 15 34 19" src="https://user-images.githubusercontent.com/1682452/180797950-b4e2f054-55a7-47aa-96f5-a9b5c9c2431e.png">
<img width="427" alt="Screenshot 2022-07-25 at 15 34 22" src="https://user-images.githubusercontent.com/1682452/180797952-aa2c78af-75dd-402f-9698-98d610935f63.png">

In the Post Editor, the 'Save draft' / 'Save' button tooltip shows only the keyboard shortcut, which is a bit odd:

<img width="167" alt="Screenshot 2022-07-25 at 15 43 47" src="https://user-images.githubusercontent.com/1682452/180798742-49fd0c0b-5d00-4eaf-bb9f-9856a53f97d1.png">

Instead, in the Site Editor, the 'Save' button doesn't show the keyboard shortcut.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Additionally: when 'Show button text labels' is enabled, most of the top toolbar buttons should be `tertiary` buttons. 

In the Post Editor, they are tertiary except 'List View'. Looking a bit into the project history, the `variant` prop was inadvertently removed in some refactoring.

Post Editor: the 'List View' text color should be blue. Note: Undo and Redo become blue when there are undo/redo steps.

<img width="1282" alt="post editor" src="https://user-images.githubusercontent.com/1682452/181701354-f2ab6a3b-570d-4b9e-89c0-116b6238e00b.png">

In the Site Editor, a few buttons aren't styled as tertiary. For consistency, they should be tertiary. For example: in the Post Editor, 'Settings` and `Options` do have blue text. In the Site Editor, they don't:

<img width="1281" alt="site editor" src="https://user-images.githubusercontent.com/1682452/181702726-be77e241-d129-4044-ae2c-d34f21fe65f8.png">

The tertiary style is determined by the `showIconLabels` prop, which isn't even passed to some of these buttons. Will fix it.
